### DISCUSSION TITLE
On plaintext TCP, arrival rate can reach 700/s, but only 100/s on TLS

### DISCUSSION THREAD

**Discussion by CH1NL1ANG** [emqx/emqx#4611] [Q&A] [answered]
测试非加密连接时，可以支持每秒建立700个连接；加密连接时只能支持每秒100个连接，请问有什么办法可以改善吗？

**Comment by CH1NL1ANG** [comment]
问题表现为，测试加密连接时，每秒超过100个连接，服务器的CPU会飚到非常高，客户端这边提示connect error

**Comment by yogkang** [comment]
SSL 的连接可定是会耗 CPU 。我想了解下你那边是 tcp 最高压测出来的每秒连接数是多少。

**Comment by zmstone** [comment, marked as answer]
跟TCP相比，TLS 的握手，加解密都会消耗更多资源。可以考虑在前面放一个LB或者proxy，把TLS断在那里
这样可以让后端（EMQ X）专注处理MQTT的业务。

Comparing to TCP, TLS handshakes requires more resource,
We recommend TLS termination by putting an LB or proxy in front of EMQ X
so EMQ X nodes can dedicate its resource to handle MQTT
